runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/compression/GzipJVM.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.compression
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.content.ByteStream
+import aws.smithy.kotlin.runtime.io.GzipByteReadChannel
+import aws.smithy.kotlin.runtime.io.GzipSdkSource
+import aws.smithy.kotlin.runtime.io.SdkByteReadChannel
+import aws.smithy.kotlin.runtime.io.SdkSource
+import java.io.ByteArrayOutputStream
+import java.util.zip.GZIPOutputStream
+
+/**
+ * The gzip compression algorithm.
+ * Used to compress http requests.
+ *
+ * See: https://en.wikipedia.org/wiki/Gzip
+ */
+@InternalApi
```

Comment:
question: why is this `InternalApi`, isn't it customer-facing?

Reply:
I think I got a comment saying this should be `InternalApi`. I don't think customers should be using this, but I don't know if that goes with the definition of the `InternalApi` annotation. "This API is internal to the smithy-kotlin runtime and should not be used". Or if actually should be used by clients

Reply:
By default we are adding the Gzip compression algorithm to the clients' supported compression algorithms list. Since we're adding it to clients, I believe it's not an internal API. I'm interested what other people think

Reply:
I agree with Matas. This is a value users may want to add to their own lists of supported compressions. It shouldn't be `@InternalApi`.